radareorg/cutter
# Seperate Widget showing grid of bytes from rest of the hexdump panel.

**Detailed description**

Split HexDumpWidget in to two parts so that one of them can be later replaced by #1494. This PR contains only refactoring and shouldn't change behavior in significant way.

The new HexTextView class only shows grid of bytes. Synchronization with other widgets and information about selected range remains in HexdumpWidget.

setupFonts handling is a bit ugly because Qt font propagation resets font set in HexTextView when it gets added to parent. Doesn't matter too much as hopefully HexTextView will get removed soon, but if anyone understands how Qt font propagation works and can suggest better solution I'd be happy to hear it.

Before this PR can be reviewed need to create test plan, test that HexTextView can be replaced with the hexdump widget from #1494.

**Test plan (required)**
* basic functionality
  * Shows bytes
  * Scroll to top
  * selection works
    * selecting bytes
    * selecting text
    * synchronization between selecting text and bytes
    * select forwards and backwards
  * copy bytes
  * copy text
  * copy address (menu and shortcut)
  * changing display mode
* right panel
  * selection gets updated when selecting
  * correct range (check using one of byte printers)
  * changing display mode works
* seek synchronization
  * seek in the colored bar at top
  * function list
  * disassm
  * seek in hexdump changes global seek
  * sync/unsync in context menu disables seek synchronization both ways
* setting changes
  * font change
  * color change
